From 7731133fe1266d1722cad212fd8750ca5e324c58 Mon Sep 17 00:00:00 2001
From: Don Armstrong <don@debian.org>
Date: Tue, 13 Nov 2012 13:04:01 -0800
Subject: [PATCH 2/2] * Breaks ca-certificates-java (<<20121112+nmu1);
 partially fixing #537051. * Provide
 update-ca-certificates and update-ca-certificates-fresh
   triggers. * Call the triggers using no-await so that
 the configuration files from   the newer version of
 ca-certificates-java are in places before the  
 upgrade. Closes: #537051.

---
 debian/ca-certificates.triggers |    4 ++++
 debian/changelog                |   12 ++++++++++++
 debian/control                  |    2 ++
 debian/postinst                 |   21 +++++++++++++++++++--
 4 files changed, 37 insertions(+), 2 deletions(-)
 create mode 100644 debian/ca-certificates.triggers

diff --git a/debian/ca-certificates.triggers b/debian/ca-certificates.triggers
new file mode 100644
index 0000000..14dec6e
--- /dev/null
+++ b/debian/ca-certificates.triggers
@@ -0,0 +1,4 @@
+interest-noawait update-ca-certificates
+interest-noawait update-ca-certificates-fresh
+interest update-ca-certificates
+interest update-ca-certificates-fresh
diff --git a/debian/changelog b/debian/changelog
index 861abed..a59fed5 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,15 @@
+ca-certificates (20121112+nmu1) unstable; urgency=low
+
+  * Non-maintainer upload
+  * Breaks ca-certificates-java (<<20121112+nmu1); partially fixing #537051.
+  * Provide update-ca-certificates and update-ca-certificates-fresh
+    triggers.
+  * Call the triggers using no-await so that the configuration files from
+    the newer version of ca-certificates-java are in places before the
+    upgrade. Closes: #537051.
+
+ -- Don Armstrong <don@debian.org>  Mon, 12 Nov 2012 15:58:11 -0800
+
 ca-certificates (20121105) unstable; urgency=low
 
   * Update mozilla/certdata.txt to version 1.86  Closes: #683728
diff --git a/debian/control b/debian/control
index 5ef776e..8f84573 100644
--- a/debian/control
+++ b/debian/control
@@ -13,9 +13,11 @@ Vcs-Browser: http://git.debian.org/?p=collab-maint/ca-certificates.git
 
 Package: ca-certificates
 Architecture: all
+Pre-Depends: dpkg (>= 1.16.1)
 Depends: openssl (>= 1.0.0), ${misc:Depends}
 Enhances: openssl
 Multi-Arch: foreign
+Breaks: ca-certificates-java (<<20121112+nmu1)
 Description: Common CA certificates
  This package includes PEM files of CA certificates to allow SSL-based
  applications to check for the authenticity of SSL connections.
diff --git a/debian/postinst b/debian/postinst
index 198c57e..9964e27 100644
--- a/debian/postinst
+++ b/debian/postinst
@@ -142,12 +142,29 @@ EOF
 	# fix bogus symlink to ca-certificates.crt on upgrades; see
 	# Debian #643667; drop after wheezy
 	if dpkg --compare-versions "$2" lt-nl 20111025; then
-	    update-ca-certificates --fresh
+	    dpkg-trigger --no-await update-ca-certificates-fresh
 	else
-	    update-ca-certificates
+	    dpkg-trigger --no-await update-ca-certificates
 	fi
     ;;
 
+    triggered)
+	for trigger in $2; do
+	    case "$trigger" in
+		update-ca-certificates)
+		    update-ca-certificates
+		    ;;
+		update-ca-certificates-fresh)
+		    update-ca-certificates --fresh
+		    ;;
+		*)
+		    echo "postinst called with unknown trigger \`$2'">&2
+		    exit 1
+		    ;;
+	    esac;
+	done;
+	;;
+	    
     abort-upgrade|abort-remove|abort-deconfigure)
 
     ;;
-- 
1.7.10.4

